### PR TITLE
unique_fd is no longer unique_resource

### PR DESCRIPTION
**Pull request by dimitry-**
unique_fd can no longer be reset to another (int) fd. In can only be
constructed or moved from anther unique_fd.

Test: run OrbitBase unit-tests

### CODE REVIEW

File: src/OrbitBase/include/OrbitBase/File.h [hunk 1]
```diff
@@ -13,11 +13,52 @@
...
+  ~unique_fd() noexcept { release(); }
+
+  unique_fd(const unique_fd&) = delete;
+  unique_fd& operator=(const unique_fd&) = delete;
+
+  constexpr unique_fd(unique_fd&& other) noexcept : fd_{other.fd_} { other.fd_ = kInvalidFd; }
+
+  unique_fd& operator=(unique_fd&& other) noexcept {
+    reset(other.fd_);
+    other.fd_ = kInvalidFd;
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/File.h:40]: I think you need to guard againat self-move!

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/OrbitBase/include/OrbitBase/File.h (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/include/OrbitBase/File.h:40]: right, thanks!

Comment (dimitry-) [reply src/OrbitBase/include/OrbitBase/File.h:42]: done, and added a test.

**Pull request merged**